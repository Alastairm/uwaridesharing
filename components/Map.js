import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native'
import MapView from 'react-native-maps';

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  map: {
    ...StyleSheet.absoluteFillObject,
  },
});

export default class Map extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      region: {
        latitude: -31.980101,
        longitude: 115.818650,
        latitudeDelta: 0.0045,
        longitudeDelta: 0.006,
      }
    };
    this.onRegionChange = this.onRegionChange.bind(this);
  }
  onRegionChange(region) {
    this.setState({region: region});
  }
  render() {
    const { region } = this.props;
    console.log(region);

    return (
      <View style={styles.container}>
        <MapView
          style={styles.map}
          region={this.state.region}
          onRegionChange={this.onRegionChange}
        >
        </MapView>
      </View>
    );
  }
}
