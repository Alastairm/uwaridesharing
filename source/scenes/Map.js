import React, { Component } from 'react';
import { AsyncStorage, StyleSheet, TextInput, Text, View } from 'react-native';

import Button from '../components/Button.js';
import Icon from 'react-native-vector-icons/FontAwesome';
import MapView from 'react-native-maps';

import Spatula from '../apis/spatula.js';
import LocationSearch from '../components/LocationSearch.js'
import Styles from './Styles.js';

import FareEstimation from './FareEstimation.js';

export default class Map extends React.Component {
  constructor(props) {
    super(props);
    this.spatula = new Spatula();
    this.state = {
      region: {
        latitude: -31.980101,
        longitude: 115.818650,
        latitudeDelta: 0.0045,
        longitudeDelta: 0.006,
      },
    };
    this.onNext = this.onNext.bind(this);
    this.onRegionChange = this.onRegionChange.bind(this);
    this.onLocationSeach = this.onLocationSeach.bind(this);
    this.slug = this.slug.bind(this);
  }
  async slug() {
    let data = await this.spatula.slugVendible();
    await AsyncStorage.setItem('spatula.submit.vendible', String(data.id));
  }
  async onNext() {
    await this.slug();
    await AsyncStorage.setItem('endpoint.lat', String(this.state.region.latitude));
    await AsyncStorage.setItem('endpoint.lon', String(this.state.region.longitude));
    this.props.navigator.push({
      component: FareEstimation
    });
  }
  onRegionChange(region) {
    this.setState({region: region});
  }
  onLocationSeach(data, details){
    this.setState({data: {location: details.geometry.location}});
    let region = this.state.region
    region.latitude = parseFloat(details.geometry.location.lat);
    region.longitude = parseFloat(details.geometry.location.lng);
    this.setState({region: region})
  }
  render() {
    const { region } = this.props;
    console.log(region);

    return (
      <View style={Styles.mapScene}>
        <MapView
          style={{...StyleSheet.absoluteFillObject}}
          region={this.state.region}
          onRegionChange={this.onRegionChange}
          >
        </MapView>
        <Icon name="map-marker" size={50} color="#0060C0" />
        <View style={Styles.scene}>
          <View style={Styles.header}>
            <LocationSearch onPress={this.onLocationSeach}/>
          </View>
          <View style={Styles.footer}>
            <Button
              backgroundColor={'#0060C0'}
              onPress={this.onNext}>
              Set Pickup Location
            </Button>
          </View>
        </View>
      </View>
    );
  }
}
