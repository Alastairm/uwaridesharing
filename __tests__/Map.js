import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Map from '../source/scenes/Map.js';

// Need to mock react-native-maps.
jest.mock('react-native-maps', () => 'MapView');

test('Matches snapshot', () => {
  const render = renderer.create(
    <Map />,
  ).toJSON();
  expect(render).toMatchSnapshot();
});
