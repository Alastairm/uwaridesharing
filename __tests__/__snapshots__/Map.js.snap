// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<View
  style={undefined}
>
  <MapView
    onRegionChange={[Function]}
    region={
      Object {
        "latitude": -31.980101,
        "latitudeDelta": 0.0045,
        "longitude": 115.81865,
        "longitudeDelta": 0.006,
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#0060C0",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ÔÅÅ
  </Text>
  <View
    style={undefined}
  >
    <View
      style={undefined}
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#C9C9CE",
                "borderBottomColor": "#b5b5b5",
                "borderBottomWidth": 0.5,
                "borderTopColor": "#7e7e7e",
                "borderTopWidth": 0.5,
                "flexDirection": "row",
                "height": 44,
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoFocus={false}
            clearButtonMode="while-editing"
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Search"
            placeholderTextColor="#A8A8A8"
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 5,
                  "flex": 1,
                  "fontSize": 15,
                  "height": 28,
                  "marginLeft": 8,
                  "marginRight": 8,
                  "marginTop": 7.5,
                  "paddingBottom": 4.5,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                  "paddingTop": 4.5,
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 60,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#0060C0",
              "borderRadius": 4,
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "opacity": 1,
              "overflow": "hidden",
              "width": 180,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007aff",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                null,
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                null,
              ]
            }
          >
            Set Pickup Location
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
